### ISSUE TITLE
Question about DFU vs DfuSe

### ISSUE THREAD

**Issue by iddq** [sourcebox/dfu-buddy#4]
Plain DFU is not supported yet, only DfuSe devices like STM32.

Hello,

Can you help me to find some references please what is the difference between DFU and DfuSe?

Thank you.

**Comment by sourcebox** [comment]
A DFU file is just a raw image that has a 16 byte suffix appended. The format of the suffix is described in Appendix B of https://www.usb.org/sites/default/files/DFU_1.1.pdf. Since the file does not specify any information about memory layout, it can only be transferred as whole, leaving it up to the device to process it.

A DfuSe file is a DFU file with the mentioned suffix, but the image itself has a specific format. This allows to define several sections at different memory addresses. The format is described in document UM0391 from ST, unfortunately there's currently no direct link to it on the ST site. You can try http://rc.fdr.hu/UM0391.pdf as alternative.

**Comment by JohnAZoidberg** [comment]
I found that I can add a DFU suffix with this tool: https://dfu-util.sourceforge.net/dfu-suffix.1.html
Then dfu-buddy will recognize it as a "Plain" DFU file.

**Comment by sourcebox** [comment]
> I found that I can add a DFU suffix with this tool: https://dfu-util.sourceforge.net/dfu-suffix.1.html Then dfu-buddy will recognize it as a "Plain" DFU file.

Yes, DFU buddy will recognize the file correctly. But it can't be flashed because plain files require a different handling on the USB layer - which is currently not implemented.

**Comment by BitLoose** [comment]
Nice looking GUI!  Has this been tested on STM32H743VIT MCUs?

**Comment by sourcebox** [comment]
> Nice looking GUI! Has this been tested on STM32H743VIT MCUs?

I think I did some tests with a H723 series chip. In general, it should work with the H7, so just give it a try. If you don't get any error messages when flashing, chances are good that everything works correctly.

**Comment by BitLoose** [comment]
Thanks, I've never used Rust before, so it might take a while to setup, I'm a C/C++ and ASM coder, not much difference hopfully

**Comment by sourcebox** [comment]
> Thanks, I've never used Rust before, so it might take a while to setup, I'm a C/C++ and ASM coder, not much difference hopfully

Rust setup is quite simple compared to C++, especially on Linux and macOS. Just use `rustup` as described on the official site. After that, all you have to do is clone the repository and run `cargo run --release`.

But I think it's also time to put some updated binaries onto the releases page.

**Comment by BitLoose** [comment]
I'm needing to add a format converter to your code to support our binary format, so I'd need to install Rust on Windows recompile.Hopefully it will be easy enoughOn 6 Mar 2025 19:58, Oliver Rockstedt ***@***.***> wrote:
  
Thanks, I've never used Rust before, so it might take a while to setup, I'm a C/C++ and ASM coder, not much difference hopfully

Rust setup is quite simple compared to C++, especially on Linux and macOS. Just use rustup as described on the official site. After that, all you have to do is clone the repository and run cargo run --release.
But I think it's also time to put some updated binaries onto the releases page.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>



sourcebox left a comment (sourcebox/dfu-buddy#4)

Thanks, I've never used Rust before, so it might take a while to setup, I'm a C/C++ and ASM coder, not much difference hopfully

Rust setup is quite simple compared to C++, especially on Linux and macOS. Just use rustup as described on the official site. After that, all you have to do is clone the repository and run cargo run --release.
But I think it's also time to put some updated binaries onto the releases page.
